src/tokens/ERC20RebaseDistributor.sol

```diff
@@ -98,8 +99,11 @@ abstract contract ERC20RebaseDistributor is ERC20 {
         uint256 balance = ERC20.balanceOf(account);
         uint256 shares = (balance * START_REBASING_SHARE_PRICE) /
             rebasingSharePrice;
-        rebasingState[account].nShares = uint248(shares);
-        rebasingState[account].isRebasing = 1;
+        rebasingState[account] = RebasingState({
+            isRebasing: 1,
+            nShares: uint136(shares),
+            minBalance: uint112(balance)
```

Comment:
Just a quick sanity check on magnitudes here:

2^136 ~ 10^40
2^112 ~ 10^33

The number of shares is regularly multiplied by a value that is >= 10^30 (I'm referring to `rebasingSharePrice`). But, share quantities themselves shrink for a given number of tokens as `rebasingSharePrice` increases by the same order of magnitude. So there isn't a big risk of overflow AFAICT.

Most 18-decimal tokens have a total supply that is less than max(uint112) ~ 10^33. Balancer v2 also successfully uses 112-bit unsigned integers to store balances. So again the cast is probably safe (although I might use a checked cast out of paranoia).